### PR TITLE
[CIRC-1894] Improve logging circulation resources: A - D

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Resolves: [CIRC-1894](https://issues.folio.org/browse/CIRC-1894)

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/AbstractCirculationRulesEngineResource.java [hunk 1]
```diff
@@ -72,13 +72,16 @@ public void register(Router router) {
...
     final String regex = "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[1-5][a-fA-F0-9]{3}-[89abAB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$";
     String uuid = request.getParam(paramName);
     if (uuid == null) {
+      log.debug("invalidUuid:: {} is null", paramName);
       ClientErrorResponse.badRequest(request.response(), "required query parameter missing: " + paramName);
       return true;
     }
-    if (! uuid.matches(regex)) {
+    if (!uuid.matches(regex)) {
+      log.debug("invalidUuid:: {} is not matched to the regex {}", uuid, regex);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/AbstractCirculationRulesEngineResource.java:84]:
```suggestion
      log.debug("invalidUuid:: {} does not match the regex {}", uuid, regex);
```

File: src/main/java/org/folio/circulation/resources/AddInfoResource.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public void register(Router router) {
   }
 
   private void addPatronOrStaffInfo(RoutingContext routingContext) {
-    log.info("Add-info invoked");
+    log.debug("addPatronOrStaffInfo:: ");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/AddInfoResource.java:49]:
```suggestion
    log.debug("addPatronOrStaffInfo:: invoked");
```

File: src/main/java/org/folio/circulation/resources/AddInfoResource.java [hunk 3]
```diff
@@ -91,25 +92,35 @@ private Result<AddInfoRequest> createAddInfoRequest(RoutingContext routingContex
...
+    return loanResult.map(l -> addPatronOrStaffInfo(l, request.getAction(),
+      request.getActionComment()));
   }
 
   private LoanAndRelatedRecords addPatronOrStaffInfo(LoanAndRelatedRecords loanAndRelatedRecords,
-                                                     String action, String actionComment) {
+    String action, String actionComment) {
+
+    log.debug("addPatronOrStaffInfo:: parameters loanAndRelatedRecords: {}, action: {}, " +
+      "actionComment: {}", loanAndRelatedRecords, action, actionComment);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/AddInfoResource.java:119]: logging `loanAndRelatedRecords` converts a loan JSON-representation into a String. Are we sure we want the whole loan in the logs? Also, it's a rather costly operation, so we better use Supplier or check if debug-level is enabled.

File: src/main/java/org/folio/circulation/resources/AddInfoResource.java [hunk 4]
```diff
@@ -91,25 +92,35 @@ private Result<AddInfoRequest> createAddInfoRequest(RoutingContext routingContex
...
     loanAndRelatedRecords.getLoan().changeAction(action);
     loanAndRelatedRecords.getLoan().changeActionComment(actionComment);
+    log.info("addPatronOrStaffInfo:: result: {}", loanAndRelatedRecords);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/AddInfoResource.java:122]: Again, do we really want to log the whole loan?

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 5]
```diff
@@ -64,6 +64,8 @@ private void changeDueDate(RoutingContext routingContext) {
   private CompletableFuture<Result<LoanAndRelatedRecords>> processChangeDueDate(
     final ChangeDueDateRequest request, RoutingContext routingContext) {
 
+    log.debug("processChangeDueDate:: parameters request: {}", request);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:67]: `ChangeDueDateRequest` does not override `toString()`

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 6]
```diff
@@ -105,6 +107,8 @@ private CompletableFuture<Result<LoanAndRelatedRecords>> processChangeDueDate(
   private LoanAndRelatedRecords unsetDueDateChangedByRecallIfNoOpenRecallsInQueue(
       LoanAndRelatedRecords loanAndRelatedRecords) {
 
+    log.debug("unsetDueDateChangedByRecallIfNoOpenRecallsInQueue:: parameters loanAndRelatedRecords: {}",
+      loanAndRelatedRecords);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:111]: same here

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 7]
```diff
@@ -117,29 +121,41 @@ private LoanAndRelatedRecords unsetDueDateChangedByRecallIfNoOpenRecallsInQueue(
     }
   }
 
-  CompletableFuture<Result<Loan>> getExistingLoan(LoanRepository loanRepository, ChangeDueDateRequest changeDueDateRequest) {
+  CompletableFuture<Result<Loan>> getExistingLoan(LoanRepository loanRepository,
+    ChangeDueDateRequest changeDueDateRequest) {
+
+    log.debug("getExistingLoan:: parameters changeDueDateRequest: {}", changeDueDateRequest);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:127]: `ChangeDueDateRequest` does not override `toString()`

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 8]
```diff
@@ -117,29 +121,41 @@ private LoanAndRelatedRecords unsetDueDateChangedByRecallIfNoOpenRecallsInQueue(
...
     return loanRepository.getById(changeDueDateRequest.getLoanId())
-      .thenApplyAsync(r -> r.map(exitingLoan -> exitingLoan.setPreviousDueDate(exitingLoan.getDueDate())));
+      .thenApplyAsync(r -> r.map(exitingLoan -> exitingLoan.setPreviousDueDate(
+        exitingLoan.getDueDate())));
   }
 
   private Result<LoanAndRelatedRecords> changeDueDate(Result<LoanAndRelatedRecords> loanResult,
       ChangeDueDateRequest request) {
 
+    log.debug("changeDueDate:: parameters request: {}", request);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:137]: `ChangeDueDateRequest` does not override `toString()`

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 9]
```diff
@@ -117,29 +121,41 @@ private LoanAndRelatedRecords unsetDueDateChangedByRecallIfNoOpenRecallsInQueue(
...
+
     return loanResult.map(l -> changeDueDate(l, request.getDueDate()));
   }
 
   private LoanAndRelatedRecords changeDueDate(LoanAndRelatedRecords loanAndRelatedRecords,
     ZonedDateTime dueDate) {
 
+    log.debug("changeDueDate:: parameters loanAndRelatedRecords: {}, dueDate: {}",
+      loanAndRelatedRecords, dueDate);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:146]: same here

File: src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java [hunk 10]
```diff
@@ -117,29 +121,41 @@ private LoanAndRelatedRecords unsetDueDateChangedByRecallIfNoOpenRecallsInQueue(
...
     loanAndRelatedRecords.getLoan().changeDueDate(dueDate);
+
     return loanAndRelatedRecords;
   }
 
   private Result<ChangeDueDateRequest> createChangeDueDateRequest(RoutingContext routingContext) {
     final String loanId = routingContext.pathParam("id");
     final JsonObject body = routingContext.getBodyAsJson();
+    log.debug("createChangeDueDateRequest:: parameters loanId: {}, body: {}", loanId, body);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java:155]: Supplier or check logging level

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 11]
```diff
@@ -170,6 +178,8 @@ CompletableFuture<Result<RequestQueue>> getRequestQueue(CheckInContext context)
   }
 
   CompletableFuture<Result<Item>> updateItem(CheckInContext context) {
+    log.debug("updateItem:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:181]: This logs both the loan and `CheckInByBarcodeRequest`. Do we need the full loan in the logs? Can we use Supplier or check logging level?

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 12]
```diff
@@ -178,6 +188,7 @@ CompletableFuture<Result<Item>> updateItem(CheckInContext context) {
   CompletableFuture<Result<RequestQueue>> updateRequestQueue(
     CheckInContext context) {
 
+    log.debug("updateRequestQueue:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:191]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 13]
```diff
@@ -186,13 +197,15 @@ CompletableFuture<Result<RequestQueue>> updateRequestQueue(
   }
 
   CompletableFuture<Result<Loan>> updateLoan(CheckInContext context) {
+    log.debug("updateLoan:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:200]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 14]
```diff
@@ -186,13 +197,15 @@ CompletableFuture<Result<RequestQueue>> updateRequestQueue(
...
     // Loan must be updated after item
     // due to snapshot of item status stored with the loan
     // as this is how the loan action history is populated
     return loanRepository.updateLoan(context.getLoan());
   }
 
   CompletableFuture<Result<Item>> getDestinationServicePoint(CheckInContext context) {
+    log.debug("getDestinationServicePoint:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:208]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 15]
```diff
@@ -206,12 +219,15 @@ CompletableFuture<Result<Item>> getDestinationServicePoint(CheckInContext contex
   }
 
   CompletableFuture<Result<ServicePoint>> getCheckInServicePoint(CheckInContext context) {
+    log.debug("getCheckInServicePoint:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:222]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 16]
```diff
@@ -206,12 +219,15 @@ CompletableFuture<Result<Item>> getDestinationServicePoint(CheckInContext contex
...
     return servicePointRepository.getServicePointById(context.getCheckInServicePointId());
   }
 
   CompletableFuture<Result<Request>> getPickupServicePoint(CheckInContext context) {
+    log.debug("getPickupServicePoint:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:227]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 17]
```diff
@@ -221,8 +237,10 @@ CompletableFuture<Result<Request>> getPickupServicePoint(CheckInContext context)
   }
 
   CompletableFuture<Result<Request>> getRequester(CheckInContext context) {
+    log.debug("getRequester:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:240]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 18]
```diff
@@ -231,8 +249,10 @@ CompletableFuture<Result<Request>> getRequester(CheckInContext context) {
   }
 
   CompletableFuture<Result<Request>> getAddressType(CheckInContext context) {
+    log.debug("getAddressType:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:252]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 19]
```diff
@@ -250,21 +270,29 @@ CheckInContext setInHouseUse(CheckInContext checkInContext) {
   public CompletableFuture<Result<CheckInContext>> logCheckInOperation(
     CheckInContext checkInContext) {
 
+    log.debug("logCheckInOperation:: parameters checkInContext: {}", checkInContext);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:273]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 20]
```diff
@@ -250,21 +270,29 @@ CheckInContext setInHouseUse(CheckInContext checkInContext) {
...
+
     return logCheckInService.logCheckInOperation(checkInContext);
   }
 
   CompletableFuture<Result<CheckInContext>> createOverdueFineIfNecessary(
     CheckInContext records, WebContext context) {
 
+    log.debug("createOverdueFineIfNecessary:: parameters records: {}", records);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:281]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 21]
```diff
@@ -250,21 +270,29 @@ CheckInContext setInHouseUse(CheckInContext checkInContext) {
...
+
     return overdueFineService.createOverdueFineIfNecessary(records, context.getUserId())
       .thenApply(r -> r.next(action -> feeFineScheduledNoticeService.scheduleOverdueFineNotices(records, action)));
   }
 
   CompletableFuture<Result<CheckInContext>> refundLostItemFees(CheckInContext context) {
+    log.debug("refundLostItemFees:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:288]: same here

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 22]
```diff
@@ -250,21 +270,29 @@ CheckInContext setInHouseUse(CheckInContext checkInContext) {
...
+
     return lostItemFeeRefundService.refundLostItemFees(context);
   }
 
   CompletableFuture<Result<CheckInContext>> findFulfillableRequest(CheckInContext context) {
+    log.debug("findFulfillableRequest:: parameters context: {}", context);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java:294]: same here

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**